### PR TITLE
Updates to HTTP service to handle urls

### PR DESCRIPTION
**Pull request by spillai**
## Checks

- [x] `make lint`: I've run `make lint` to lint the changes in this PR.
- [x] `make test`: I've made sure the tests (`make test-cpu` or `make test`) are passing.
- Additional tests:
   - [ ] Benchmark tests (when contributing new models)
   - [ ] GPU/HW tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ image, = sdxl(prompts=["fox jumped over the moon"],
 
 ```bash
 curl \
--X POST http://localhost:8000/infer \
+-X POST http://localhost:8000/v1/infer \
```
Comment (outtanames) [inline README.md:90]: :100: we're a real API now

File: nos/server/http/_service.py [hunk 2]
```diff
@@ -87,7 +88,7 @@ def is_video(filename: str) -> bool:
 
 def is_audio(filename: str) -> bool:
     """Check if the given filename is an audio."""
-    return filename.lower().endswith((".mp3", ".wav", ".flac", ".m4a"))
+    return filename.lower().endswith((".mp3", ".mp4", ".wav", ".flac", ".m4a", ".webm"))
```
Comment (outtanames) [inline nos/server/http/_service.py:91]: [nit] should probably move this to a `SUPPORTED_AUDIO_EXTENSIONS` const

File: nos/server/http/_service.py [hunk 3]
```diff
@@ -119,40 +120,111 @@ def get_client() -> Client:
...
+            -F 'model_id=yolox/small' \
+            -F 'url=https://.../test.jpg'
+
+        """
+        logger.debug(f"Received inference request [model={model_id}, method={method}]")
+        request = InferenceRequest(model_id=model_id, method=method, inputs={})
+        file_object: SpooledTemporaryFile = None
+        file_basename: str = None
+        if file is not None:
+            assert isinstance(file.file, SpooledTemporaryFile), f"Invalid file [file={file}]"
```
Comment (outtanames) [inline nos/server/http/_service.py:197]: [nit] might eventually want to check the filesize here and set a reasonable maximum

**Review by outtanames — APPROVED** [review]
LGTM

File: nos/server/http/_service.py (hunk 2 above)
Comment (spillai) [reply nos/server/http/_service.py:91]: Agreed, will add these on another refactor PR.

File: nos/server/http/_service.py (hunk 3 above)
Comment (spillai) [reply nos/server/http/_service.py:197]: Agreed. Will probably add a bunch of these checks on the HTTP gateway side.

**Pull request merged**